### PR TITLE
SizeObserver crusade: Dismissable

### PR DESCRIPTION
**Pull request by Hixie**
Refactor Dismissable to not need a SizeObserver and to generally tighten
up the code a bit.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth, @HansMuller

File: packages/flutter/lib/src/widgets/dismissable.dart [hunk 1]
```diff
@@ -68,86 +68,59 @@ class Dismissable extends StatefulComponent {
 class _DismissableState extends State<Dismissable> {
   void initState() {
     super.initState();
-    _dismissController = new AnimationController(duration: _kCardDismissDuration);
-    _dismissController.addStatusListener((AnimationStatus status) {
-      if (status == AnimationStatus.completed)
-        _handleDismissCompleted();
-    });
+    _moveController = new AnimationController(duration: _kCardDismissDuration);
+    _moveController.addStatusListener(_handleDismissStatusChanged);
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/dismissable.dart]: I usually use the `..` operator here.

File: packages/flutter/lib/src/widgets/dismissable.dart [hunk 2]
```diff
@@ -68,86 +68,59 @@ class Dismissable extends StatefulComponent {
...
-      _resizeController.forward();
-    });
-  }
-
-  void _handleResizeProgressChanged() {
-    if (_resizeController.isCompleted)
-      _maybeCallOnDismissed();
-    else
-      _maybeCallOnResized();
+  Size findSize() {
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/dismissable.dart]: _findSize

**Comment by abarth** [comment]
LGTM

**Pull request merged**